src/components/Question.js

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+
+const Question = ({ question, answers, correct, id, checked, setSuccess }) => {
+  const [submitted, setSubmitted] = useState(null);
+
+  const responseElement = answers.map((response, index) => (
+    <div
+      key={index}
+      className={`border whitespace-nowrap border-primary-100  rounded-lg px-2 py-1 ${
+        submitted === index && !checked && 'bg-indigo-200'
+      } ${response === correct && checked && 'bg-green-300'} ${
+        submitted === index && correct !== response && checked && 'bg-red-100'
+      } `}
+    >
+      <button
+        disabled={checked}
+        onClick={() => {
+          setSubmitted(index);
+          setSuccess(prev => {
+            prev[id] = response === correct;
+            return prev;
+          });
+        }}
+      >
+        {response}
+      </button>
+    </div>
+  ));
+
+  return (
+    <div className='pt-10 pb-3 border-b-2 w-9/12'>
+      <h1 className='text-xl font-semibold' dangerouslySetInnerHTML={{ __html: question }} />
```

Comment:
From what I know, The `dangerouslySetInnerHTML` prop is used to set HTML content directly, which can be a security risk if the content is not trusted. It should be used with caution and only when necessary. 
That being said, I believe you should avoid using `dangerouslySetInnerHTML`. Instead, you can simply use the `question` variable as the content of the h1 element. Something like this: 
``` suggestion
     <h1 className='text-xl font-semibold'>{question}</h1>
```

Reply:
Okay, I understand. Maybe I will try to find another way to decode text. In the meantime, the way I saw involved using dangerouslySetInnerHTML. However, if I try to implement the way you suggested to me, there may be instances where special characters are involved, which could make it difficult for the user to read the question. what do you thing?

Reply:
Oh, in that case maybe you can use the regex, but in the meantime, we can continue with the current approach.